### ISSUE TITLE
Progressive typeahead results

### ISSUE THREAD

**Issue by cdmckay** [angular-ui/bootstrap#3931]
I have an app that is integrated with another app. As such, we have two sources of typeahead suggestions: those in local memory, and those from an API endpoint.

What I'd like to do is have it so when someone uses the typeahead, it immediately shows the local results, and then merges them with API endpoint results once they arrive.

Is this possible with the typeahead directive as it stands, or will I need to modify it?

Basically, if I modify the array returned to the typeahead directive, will it be watching?

**Comment by RobJacobs** [comment]
This is not the place to ask implementation questions, https://github.com/angular-ui/bootstrap#faq.
